version: "3.8"
services:
  postgres:
    image: postgres:13.5-alpine
    container_name: postgres
    volumes:
      - "$PWD/docker-pgsql-entrypoint:/docker-entrypoint-initdb.d"
      - "$PWD/docker-pgsql-volume:/var/lib/postgresql/data" 
    ports:
      - 5432:5432
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=testing_password
      - TESTING_POSTGRES_USER=testing_user
      - TESTING_POSTGRES_DB=testing_metadb
    deploy:
      restart_policy:
        condition: on-failure
        max_attempts: 3
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 15s
      timeout: 5000s
      retries: 50
  testing:
    build:
      context: ./
    stdin_open: true
    tty: true
    container_name: testing_env
    image: terraform-aws-infrastructure-live-ci-testing:latest
    volumes:
      - "$PWD:/src"
    environment:
      - PGUSER
      - PGPASSWORD
      - PGDATABASE
      - PGHOST
      - PGPORT
      - ADDITIONAL_PATH
    command: ["/bin/bash"]
    entrypoint: ["/bin/bash", "/src/entrypoint.sh"]